crates/admin/src/cluster_controller/cluster_state_refresher.rs

```diff
@@ -166,32 +172,50 @@ impl<T: TransportConnect> ClusterStateRefresher<T> {
-                    _ => {
+                    Err(NetworkError::RemoteVersionMismatch(msg)) => {
+                        // When **this** node has just started, other peers might not have
+                        // learned about the new metadata version and then they can
+                        // return a RemoteVersionMismatch error.
+                        // In this case we are not sure about the peer state but it's
+                        // definitely not dead!
+                        // Hence we set it as Suspect node. This gives it enough time to update
+                        // its metadata, before we know the exact state
+                        debug!("Node {node_id} is marked as Suspect: {msg}");
+                        nodes.insert(
+                            node_id.as_plain(),
+                            NodeState::Suspect(SuspectNode {
+                                generational_node_id: node_id,
+                                last_attempt: MillisSinceEpoch::now(),
+                            }),
+                        );
+                    }
+                    Err(err) => {
                         // todo: implement a more robust failure detector
                         // This is a naive mechanism for failure detection and is just a stop-gap measure.
                         // A single connection error or timeout will cause a node to be marked as dead.
-                        let last_seen_alive =
-                            last_state
-                                .nodes
-                                .get(&node_id)
-                                .and_then(|state| match state {
-                                    NodeState::Alive(AliveNode {
-                                        last_heartbeat_at, ..
-                                    }) => Some(*last_heartbeat_at),
-                                    NodeState::Dead(DeadNode { last_seen_alive }) => {
-                                        *last_seen_alive
-                                    }
-                                });
-
-                        nodes.insert(node_id, NodeState::Dead(DeadNode { last_seen_alive }));
+                        trace!("Node {node_id} is marked dead {node_id}: {err}");
+                        let last_seen_alive = last_state.nodes.get(&node_id.as_plain()).and_then(
+                            |state| match state {
+                                NodeState::Alive(AliveNode {
+                                    last_heartbeat_at, ..
+                                }) => Some(*last_heartbeat_at),
+                                NodeState::Dead(DeadNode { last_seen_alive }) => *last_seen_alive,
+                                NodeState::Suspect(_) => None,
+                            },
+                        );
+
+                        nodes.insert(
+                            node_id.as_plain(),
+                            NodeState::Dead(DeadNode { last_seen_alive }),
```

Comment:
I guess an orthogonal aspect to this problem is that we consider a node dead on the first failed heartbeat which might lead to more false positives.

Reply:
So far this is true, but this will change when introducing `failure detection/gossip` to the system. Right we only consider the node as Suspect if we get a specific version mismatch error